tensorboard/components/tensor_widget/README.md

```diff
@@ -0,0 +1,21 @@
+# Tensor Widget
+
+Tensor Widget is a DOM library for flexible, interactive, and efficient
+display of values of tensors, i.e., typed n-dimensional arrays.
+
+Features
+
+- The header region displays the dtype, shape and optionally the name of
+  the tensor.
+- The **health pill** section of the header displays succinct summary of the
+  tensor's element values, e.g., minimum, maxmimum, arithmetic mean,
+  standard deviation, as well as distribution of the elements among value
+  types such as zero, finite, infinite and NaN.
+- In the main section of the basic UI, the widget displays element values as
```

Comment:
A text view is a basic view of a tensor. It is convenient when some one wants to examine the details numeric values of the tensor. This kind of need arises when, e.g., a developer is checking the mathematical correctness when implementing a new layer or op type in TF. Well-established data science GUIs such as PyCharm's data view (https://www.jetbrains.com/help/pycharm/viewing-as-array.html) have this view.

The concern about large tensor sizes is addressed by the lazy value fetch support of TensorWidget.

The more visual representation is taken into account. See the "custom renderers" section of the design doc.

Reply:
I understand there's use cases for a tabular text view and that an image-based view is also planned; it was more a question of which aspect made sense as the primary view.  The context seems a little different than an IDE like PyCharm or RStudio where the primary interaction mode is expected to be text/code-driven and visualization is secondary; I think of TensorBoard as visual first and textual secondary.

Especially since presumably TensorWidget is meant to itself be a sub-surface of more complex visualizations (e.g. within the graph viz, or the debugger) it seemed like defaulting to a more compact informational representation could be useful to keep the overall GUI streamlined.  But as long as both views are supported I guess there's room down the road to figure out which one should be primary in which context.